apps/codelab/src/app/components/buttons-nav-bar/button-with-menu/button-with-menu.component.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, ContentChild, ElementRef, ViewContainerRef } from '@angular/core';
+import { Overlay, OverlayConfig } from '@angular/cdk/overlay';
+import { ButtonWithMenuModalDirective } from './button-with-menu-modal.directive';
+import { TemplatePortal } from '@angular/cdk/portal';
+
+@Component({
+  selector: 'button-with-menu',
+  templateUrl: './button-with-menu.component.html'
+})
+export class ButtonWithMenuComponent {
+
+  @ContentChild(ButtonWithMenuModalDirective) modal;
+
+  constructor(
+    private _vcr: ViewContainerRef,
+    readonly el: ElementRef,
+    readonly overlay: Overlay
+  ) {
+  }
+
+  open() {
+
+    const positionStrategy = this.overlay.position()
+      .flexibleConnectedTo(this.el.nativeElement).withPositions([
+        {
+          originX: 'center',
+          overlayX: 'center',
+          originY: 'top',
+          overlayY: 'bottom',
+        },
+      ]);
+    const overlayConfig = new OverlayConfig({
+      hasBackdrop: true,
+      positionStrategy
+    });
+
+    const overlayRef = this.overlay.create(overlayConfig);
+    const portal = new TemplatePortal(this.modal.template, this._vcr, {
+      $implicit: {
+        close: () => {
+          overlayRef.detach();
+        }
+      }
+    });
+
+    overlayRef.attach(portal);
+    overlayRef.backdropClick().subscribe(() => overlayRef.dispose());
```

Comment:
missing unsubscribe to dispose subscription

Reply:
![image](https://user-images.githubusercontent.com/40883170/52914509-439af980-32da-11e9-8b6a-81a0ceec0b0d.png)
![image](https://user-images.githubusercontent.com/40883170/52914513-4990da80-32da-11e9-9d21-2e348872d6d5.png)

Is it right way to unsubscribe? Sorry if the question is stupid)

Reply:
It is necessary to check the subscription is released when overlay is disposed (manual and automatically). If is not released, then needed to add unsubscribe when overlay/portal are detached.

When overlay is creation, we need to check current overlay is existing. If is existing, we need to dispose old overlay.

Also it's needed in ngOnDestroy for dispose overlay/portal/subscription.